### ISSUE TITLE
CVXPY 1.0: Non-symmetric solutions in SDPs

### ISSUE THREAD

**Issue by nrontsis** [cvxgrp/cvxpy#415]
Consider the following problem:

<img width="566" alt="problem" src="https://user-images.githubusercontent.com/4145552/33584814-00293d2a-d916-11e7-8e63-5401fa4d05e0.png">

```
import cvxpy as cvx
import numpy as np

M = cvx.Variable((3, 3), PSD=True)
C1 = np.array([[0, 0, 1/2], [0, 0, 0], [1/2, 0, 1]])
C2 = np.array([[0, 0, 0], [0, 0, 1/2], [0, 1/2, 1]])
constraints = [M + C1 == cvx.Variable((3,3), PSD=True)]
constraints += [M + C2 == cvx.Variable((3,3), PSD=True)]
objective = cvx.Minimize(cvx.trace(M))
prob = cvx.Problem(objective, constraints)
opt_val = prob.solve()

# Check solution is symmetric
assert (M.value.T == M.value).all()
```
The above generates **non symmetric** solutions.

CVXPY 1.0 (9d0e961), Python 3.6.3 (via Ananconda), macOS Sierra, solvers tried MOSEK==8.1.33, SCS==2.0.2.

**Comment by nrontsis** [comment]
Closing, as I've realised that this does not occur on this on a clean environment. Sorry for the confusion.

**Comment by sschnug** [comment]
I don't have that version right now and therefore i don't know if you indeed observe wrong results or if you are just disregarding floating-point math.

    (M.value.T == M.value)  # will do a comparison of fp-values without any tolerance!

So, how exactly are you defining non symmetric?

Demo (with numpy-arrays here; should transfer to cvxpy):

    import numpy as np
    np.random.seed(1)

    a = np.random.random((3,3))
    M = a.dot(a.T)/2
    print(M)
    print('1: assert: ' + str((M.T == M).all()))

    M[0,1] += 1e-15                                       # precision typically not achieved by solvers
    print(M)
    print('2: assert: ' + str((M.T == M).all()))

    print('3: assert: ' + str((np.allclose(M.T, M))))     # good-practice fp-comparison

    [[0.34638737 0.11590088 0.16331792]
     [0.11590088 0.06073435 0.07183128]
     [0.16331792 0.07183128 0.15576476]]
    1: assert: True
    [[0.34638737 0.11590088 0.16331792]
     [0.11590088 0.06073435 0.07183128]
     [0.16331792 0.07183128 0.15576476]]
    2: assert: False
    3: assert: True

**Comment by nrontsis** [comment]
Thanks for the reply @sschnug. Sorry for the confusion, I should have used `np.allclose`. But, I indeed get visually different results:

```
print(M)
[[ 0.1740965  -0.01455408 -0.07117049]
 [-0.01455408 -0.07117049 -0.07117049]
 [ 0.1740965  -0.07117049  0.06349708]]
```
I could also try installing a commit that you have/prefer.

**Comment by SteveDiamond** [comment]
There is indeed a bug here. I've fixed it on the latest 1.0 commit.
